052-cruise-control-api-users.md

```diff
@@ -0,0 +1,204 @@
+
+#### KafkaConnect users
+
+At this time, Strimzi doesn't offer any method of creating or configuring KafkaConnect API users.
+That being said, the direction we take with creating and managing Cruise Control API users could be used as a precedent for its design in the future.
+
+## Alternatives
+
+### Cruise Control user custom resource
+
+Following the pattern used for Kafka users, we could have `CruiseControlUser` resources.
+
+This would make the management of Cruise Control API users consitent with how we manage Kafka users
+
+The spec could look somthing like this:
+
+```yaml=
+apiVersion: kafka.strimzi.io/v1beta2
+kind: CruiseControlUser
+metadata:
+  name: my-user-1
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  role: USER (3)
+  password:
+    valueFrom:
+      secretKeyRef:
+        name: my-secret (1)
+        key: my-password (2)
+```
+
+(1) The name of the secret containing the predefined password.
+
+(2) The key for the password stored inside the secret.
+
+(3) By default Cruise Control defines three roles: `VIEWER`, `USER` and `ADMIN`. 
+For more details checkout the [Cruise Control Wiki](https://github.com/linkedin/cruise-control/wiki/Security#authorization)
+```
+VIEWER role: has access to the most lightweight kafka_cluster_state, user_tasks and review_board endpoints.
+USER role: has access to all the GET endpoints except bootstrap and train.
+ADMIN role: has access to all endpoints.
+```
+
+The problem with this approach is that it would add more complexity to the code base since it would require:
+* A new Strimzi `CruiseControlUser` custom resource
+* A watch on the new `CruiseControlUser` resource
+* A reconciler to manage the `CruiseControlUser` resource.
+
+### Developer managed API user configuration
+
+Having the developer manage a single secret with their custom Cruise Control API user configuration would greatly reduce the complexity of the proposed design.
+Strimzi could add the contents of this secret to the Cruise Control API user configuration file containing Strimzi's `admin` and `healthcheck` API users.
+
+The schema could look something like this:
+
+```yaml=
+apiVersion: kafka.strimzi.io/v1beta2
+kind: Kafka
+metadata:
+  ...
+spec:
+  cruiseControl:
+    apiUsersConfig:
+      secret:
+        secretName: cruise-control-api-users-secret
+     ...
+```
+
+Although this would simplify the overall design it would also shift a lot of the maintanence responsibility to the developer.
+The developer would need to create a valid Cruise Control API user configuration and encode it into a secret.
+
+## Rejected alternatives
+
+### Move API user configuration ConfigMap
+
+Having the API users configuration exposed in a `ConfigMap` would significantly reduce the code complexity of configuration management.
+That being said, the API users configuration contains sensitive information such as passwords and therefore should not be exposed via a ConfigMap.
```

Comment:
Why is ConfigMap an option here? I think the right Kubernetes resource is Secret which will have the same complexity but not the sensitivity drawback. So I do not follow why was this considered.

Reply:
You are right, removed